core/dropdowndiv.js

```diff
@@ -166,6 +166,38 @@ Blockly.DropDownDiv.setColour = function(backgroundColour, borderColour) {
 };
 
 /**
+ * Shortcut to show and place the drop-down with positioning determined
+ * by a particular block. The primary position will be below the block,
+ * and the secondary position above the block. Drop-down will be
+ * constrained to the block's workspace.
+ * @param {Object} owner The object showing the drop-down
+ * @param {!Blockly.Block} block Block to position the drop-down around.
+ * @param {Function=} opt_onHide Optional callback for when the drop-down is hidden.
+ * @param {Number} opt_secondaryYOffset Optional Y offset for above-block positioning.
+ * @return {boolean} True if the menu rendered below block; false if above.
+ */
+Blockly.DropDownDiv.showPositionedByBlock = function(owner, block,
+      opt_onHide, opt_secondaryYOffset) {
+  var scale = block.workspace.scale;
+  var bBox = block.getHeightWidth();
+  bBox.width *= scale;
+  bBox.height *= scale;
+  var position = goog.style.getPageOffset(block.getSvgRoot());
```

Comment:
Well, not sure if this is the best structure. But the DropDownDiv is positioned absolutely outside the Blockly SVG, just like the WidgetDiv. For blockly.getSvgXY_, "The origin (0,0) is the top-left corner of the Blockly SVG." - but what I actually want here is the position of the block relative to the page. Could use blockly.getSVGXY_ and then add the position of the Blockly workspace, but this seemed simpler.